### PR TITLE
ENH add missing packages to removals

### PR DESCRIPTION
**Pull request by beckermr**
Checklist

* [x] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/README.md)).
* [x] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [x] Ran `pre-commit run -a` and ensured all files pass the linting checks.
* [x] Ran `python show_diff.py` and posted the output as part of the PR.
* [x] Modifications won't affect packages built in the future. 

This PR adds these missing packages to the removals so that we don't tell someone to try and download them.

xref: https://github.com/conda/infrastructure/issues/892#issuecomment-2008028131

### CODE REVIEW

**Comment by beckermr** [comment]
pre-commit.ci autofix

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -42,7 +42,23 @@
...
+        "ffmpeg-progress-yield-0.7.6-pyhd8ed1ab_0.conda",
+        "idds-doma-1.3.0-pyhd8ed1ab_0.conda",
+        "pandera-core-0.14.3-pyhd8ed1ab_0.conda",
+        "pandera-pyspark-0.14.3-hd8ed1ab_0.conda",
+        "perl-extutils-makemaker-7.68-pl5321hd8ed1ab_0.conda",
+        "pyhanko-0.17.2-pyhd8ed1ab_0.conda",
+        "python-flatbuffers-23.3.3-pyhd8ed1ab_0.conda",
+        "strawberry-graphql-with-asgi-0.162.0-pyhd8ed1ab_0.conda",
+    ),
+    "linux-aarch64": ("awscli-1.29.39-py38he37f277_0.conda",),
```
Comment (jakirkham) [inline recipe/gen_patch_json.py:61]: Should we wrap this to cutdown on future line noise?

```suggestion
    "linux-aarch64": (
        "awscli-1.29.39-py38he37f277_0.conda",
    ),
```

**Review by jakirkham — COMMENTED** [review]
Thanks Matt! 🙏

Looks good

Should we add a comment about the reason for the removal? Also had one style suggestion below (Idk whether pre-commit will like it)

File: recipe/gen_patch_json.py (hunk 1 above)
Comment (beckermr) [reply recipe/gen_patch_json.py:61]: Pre-commit just removed this change, so no apparently. :)

File: recipe/gen_patch_json.py [hunk 2]
```diff
@@ -42,7 +42,23 @@
 )
 
 REMOVALS = {
-    "noarch": ("sendgrid-5.3.0-py_0.tar.bz2",),
+    "noarch": (
+        "sendgrid-5.3.0-py_0.tar.bz2",
```
Comment (beckermr) [inline recipe/gen_patch_json.py:46]:
```suggestion
        # the removals starting here are failed uploads
        "sendgrid-5.3.0-py_0.tar.bz2",
```

File: recipe/gen_patch_json.py [hunk 3]
```diff
@@ -42,7 +42,23 @@
...
+        "conda-forge-pinning-2023.06.08.10.36.51-hd8ed1ab_0.conda",
+        "conda-forge-repodata-patches-20230516.19.05.39-hd8ed1ab_0.conda",
+        "ffmpeg-progress-yield-0.7.6-pyhd8ed1ab_0.conda",
+        "idds-doma-1.3.0-pyhd8ed1ab_0.conda",
+        "pandera-core-0.14.3-pyhd8ed1ab_0.conda",
+        "pandera-pyspark-0.14.3-hd8ed1ab_0.conda",
+        "perl-extutils-makemaker-7.68-pl5321hd8ed1ab_0.conda",
+        "pyhanko-0.17.2-pyhd8ed1ab_0.conda",
+        "python-flatbuffers-23.3.3-pyhd8ed1ab_0.conda",
+        "strawberry-graphql-with-asgi-0.162.0-pyhd8ed1ab_0.conda",
```
Comment (beckermr) [inline recipe/gen_patch_json.py:59]:
```suggestion
        "strawberry-graphql-with-asgi-0.162.0-pyhd8ed1ab_0.conda",
        # end of removals for failed uploads
```

File: recipe/gen_patch_json.py [hunk 4]
```diff
@@ -197,6 +213,7 @@
         "nlopt-2.4.2-0.tar.bz2",
         "pygpu-0.6.5-0.tar.bz2",
         "pytest-regressions-1.0.1-0.tar.bz2",
+        "libgz-sim8-8.1.0-h86fce40_0.conda",
```
Comment (beckermr) [inline recipe/gen_patch_json.py:216]:
```suggestion
        "libgz-sim8-8.1.0-h86fce40_0.conda",  # failed upload
```

File: recipe/gen_patch_json.py (hunk 1 above)
Comment (beckermr) [inline recipe/gen_patch_json.py:61]:
```suggestion
    "linux-aarch64": (
        "awscli-1.29.39-py38he37f277_0.conda",  # failed upload
    ),
```

Maybe pre-commit is ok if there is a comment.

File: recipe/gen_patch_json.py [hunk 5]
```diff
@@ -42,7 +42,27 @@
 )
 
 REMOVALS = {
-    "noarch": ("sendgrid-5.3.0-py_0.tar.bz2",),
+    "noarch": (
+        # the removals starting here are failed uploads
+        "sendgrid-5.3.0-py_0.tar.bz2",
+        "boto3-stubs-lite-1.26.89-pyhd8ed1ab_0.conda",
```
Comment (beckermr) [inline recipe/gen_patch_json.py:48]:
```suggestion
        "sendgrid-5.3.0-py_0.tar.bz2",
        # the removals starting here are failed uploads
        "boto3-stubs-lite-1.26.89-pyhd8ed1ab_0.conda",
```

**Comment by beckermr** [comment]
pre-commit.ci autofix

**Comment by jakirkham** [comment]
Like the block comment idea that you used. Applied in the other cases

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Matt! 🙏

Let's get this in :shipit:

**Pull request merged**